urld558a3d988_gao715108023_resourceMonitor_tgz-pJ8-com_sinopec_agent_AgentJ8/src/com/sinopec/jdbc/JDBCCommon.java

```diff
@@ -0,0 +1,84 @@
+package com.sinopec.jdbc;
+
+import com.sinopec.utils.LogUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+/**
+ * Created with IntelliJ IDEA. User: gaochuanjun Date: 13-10-29 Time: 下午5:34 To
+ * change this template use File | Settings | File Templates.
+ */
+public class JDBCCommon {
+
+	private static final Log LOG = LogFactory.getLog(JDBCCommon.class);
+
+	private String ipAddress;
+
+	private String dataBaseName;
+
+	private String user;
+
+	private String password;
+
+	public JDBCCommon(String ipAddress, String user, String password, String dataBaseName) {
+		this.ipAddress = ipAddress;
+		this.user = user;
+		this.password = password;
+		this.dataBaseName = dataBaseName;
+	}
+
+	public Connection getOrclConn() {
+
+		String url = "jdbc:oracle:thin:@" + ipAddress + ":1521:" + dataBaseName; // 连接数据库的url地址
+
+		Connection conn = null;
+		try {
+			// 加载JDBC驱动
+			Class.forName("oracle.jdbc.driver.OracleDriver").newInstance();
+			// 创建数据库连接
+			conn = DriverManager.getConnection(url, user, password);
+		} catch (Exception e) {
+			LOG.error(LogUtils.getTrace(e));
+		}
+		return conn;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 37:48, 53:63.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.